File: leetcode/palindrome-number/dart/solution1/playground.dart
void main() {
  final solution = Solution();

  final testCases = [
    {'x': 121, 'expected': true},          // palindrome
    {'x': -121, 'expected': false},        // negative
    {'x': 10, 'expected': false},          // not palindrome
    {'x': 0, 'expected': true},            // single digit, palindrome
    {'x': 1221, 'expected': true},         // even-length palindrome
    {'x': 12321, 'expected': true},        // odd-length palindrome
    {'x': 1000021, 'expected': false},     // leading zero when reversed
    {'x': 2147447412, 'expected': true},   // large palindrome near int limit
  ];

  for (var i = 0; i < testCases.length; i++) {
    final x = testCases[i]['x'] as int;
    final expected = testCases[i]['expected'] as bool;

    final result = solution.isPalindrome(x);

    print('Test case ${i + 1}:');
    print('Input: x = $x');
    print('Output: $result');
    print('Expected: $expected');
    print(result == expected ? '✅ Passed\n' : '❌ Failed\n');
  }
}

File: leetcode/palindrome-number/kotlin/solution1/playground.kt
fun main() {
    val solution = Solution()

    val testCases = listOf(
        Pair(121, true),           // palindrome
        Pair(-121, false),         // negative
        Pair(10, false),           // not palindrome
        Pair(0, true),             // single digit
        Pair(1221, true),          // even-length palindrome
        Pair(12321, true),         // odd-length palindrome
        Pair(1000021, false),      // leading zero when reversed
        Pair(2147447412, true)     // large palindrome near int limit
    )

    for ((index, testCase) in testCases.withIndex()) {
        val (x, expected) = testCase
        val result = solution.isPalindrome(x)

        println("Test case ${index + 1}:")
        println("Input: x = $x")
        println("Output: $result")
        println("Expected: $expected")
        println(if (result == expected) "✅ Passed\n" else "❌ Failed\n")
    }
}
